pkg/features/kube_features.go

```diff
@@ -428,6 +428,14 @@ const (
 	// fallback to using it's cgroupDriver option.
 	KubeletCgroupDriverFromCRI featuregate.Feature = "KubeletCgroupDriverFromCRI"
 
+	// owner: @haircommander
+	// kep: https://kep.k8s.io/3983
+	// alpha: v1.28
+	//
+	// Allows the kubelet to specify a "--config-dir" flag. All .conf files in the specified
+	// directory will be parsed into the kubelet configs upon startup, overwriting any defaults
+	KubeletDropInConfig featuregate.Feature = "KubeletDropInConfig"
```

Comment:
do we need a feature gate if all the logic is under the new flag? Same as https://github.com/kubernetes/kubernetes/pull/115220

Reply:
folks thought it was best to have it be feature gated